sdk/storage/azure-storage-blob/src/main/java/com/azure/storage/blob/models/BlobDownloadAsyncResponse.java

```diff
@@ -93,10 +92,12 @@ private static Flux<ByteBuffer> createResponseFlux(StreamResponse sourceResponse
      */
     public Mono<Void> writeValueToAsync(AsynchronousByteChannel channel, ProgressReporter progressReporter) {
         Objects.requireNonNull(channel, "'channel' must not be null");
-        if (sourceResponse != null) {
-            return IOUtils.transferStreamResponseToAsynchronousByteChannel(channel, sourceResponse, onErrorResume,
-                progressReporter, retryOptions.getMaxRetryRequests());
-        } else if (super.getValue() != null) {
+//        if (sourceResponse != null) {
+//            return IOUtils.transferStreamResponseToAsynchronousByteChannel(channel, sourceResponse, onErrorResume,
+//                progressReporter, retryOptions.getMaxRetryRequests());
+//        }
```

Comment:
Remove the commented code

Reply:
I'll add comments as we'll want to keep this once the issue is resolved in the lower levels as it'll allow us to reintroduce performance improvements.